From 1014d0d8b6f750c1de87cef2aa59b28f580ce68b Mon Sep 17 00:00:00 2001
From: ruslanbay <67730802+ruslanbay@users.noreply.github.com>
Date: Sun, 21 Dec 2025 23:04:05 +0100
Subject: [PATCH 31/31] update IPU4 options in Kconfig

---
 drivers/media/pci/intel/Kconfig | 36 +++++++++++++++------------------
 1 file changed, 16 insertions(+), 20 deletions(-)

diff --git a/drivers/media/pci/intel/Kconfig b/drivers/media/pci/intel/Kconfig
index a6a2e2165923..405e757bb1be 100644
--- a/drivers/media/pci/intel/Kconfig
+++ b/drivers/media/pci/intel/Kconfig
@@ -30,22 +30,28 @@ config VIDEO_INTEL_IPU
 	select PHYS_ADDR_T_64BIT
 	select COMMON_CLK
 	help
-		Say Y here!
+	  Main driver for Intel Image Processing Unit. Say Y here.
 
 choice
-	prompt "intel ipu generation type"
+	prompt "Intel IPU generation type"
 	depends on VIDEO_INTEL_IPU
 	default VIDEO_INTEL_IPU4P
 
 config VIDEO_INTEL_IPU4
 	bool "Compile for IPU4 driver"
 	help
-		Say Y here!
+	  Select IPU4 for Intel Celeron N3350, Pentium N4200 or
+	  Atom E3900 Series Imaging Unit (PCI ID 8086:5a88).
 
 config VIDEO_INTEL_IPU4P
 	bool "Compile for IPU4P driver"
 	help
-		Say Y here!
+	  Select IPU4P for Intel Ice Lake (PCI ID 8086:8a19).
+	  Supported systems include:
+	  - Surface Pro 7
+	  - Surface Book 3
+	  - Surface Laptop 3
+	  - Dell XPS 13 7390 2-in-1
 
 endchoice
 
@@ -57,31 +63,21 @@ choice
 config VIDEO_INTEL_IPU_SOC
 	bool "Compile for SOC"
 	help
-		Select for SOC platform
+	  Intel IPU hardware platform type
 
 endchoice
 
 config VIDEO_INTEL_IPU_FW_LIB
 	bool "Compile firmware library"
+	depends on VIDEO_INTEL_IPU
 	default y
 	help
-		If selected, the firmware hostlib css would be compiled
+	  If selected, the firmware hostlib css will be compiled.
 
 config VIDEO_INTEL_IPU_WERROR
-	bool "Force GCC to throw an error instead of a warning when compiling"
-	depends on VIDEO_INTEL_IPU
-	depends on EXPERT
-	depends on !COMPILE_TEST
+	bool "Force GCC to throw errors instead of warnings for IPU"
+	depends on VIDEO_INTEL_IPU && EXPERT && !COMPILE_TEST
 	default n
 	help
-	  Add -Werror to the build flags for (and only for) intel ipu module.
-	  Do not enable this unless you are writing code for the ipu module.
-
+	  Adds -Werror to the build flags for the Intel IPU module.
 	  Recommended for driver developers only.
-
-	  If in doubt, say "N".
-
-config VIDEO_INTEL_UOS
-        bool "Compile driver per UOS"
-        help
-	        If selected UOS driver components will be compiled
-- 
2.51.0

